### DISCUSSION TITLE
How to offset page numbers to match actual page-numbers in the PDF?

### DISCUSSION THREAD

**Discussion by rudolf-adamkovic** [vedang/pdf-tools#142] [General / FAQ]
I have a number of books where the mode line shows the current page (`pdf-view-current-page`) different from the actual page number in the book.  I wonder, could I somehow instruct `pdf-tools` to offset the page number by _x_ for a particular book.  How do you folks deal with the problem?  Or, do you adjust the PDF instead?  Or do you ignore the problem?  It makes it slower to locate the page number when citing, which I do constantly (for my notes).

**Comment by orgtre** [comment]
There are `(pdf-view-current-pagelabel)` and `(pdf-view-goto-label pagelabel)` which work surprisingly well. 

You can set the mode line to anything you like, for example in plain Emacs just overwrite:
```lisp
(setq-local mode-line-position
            '(" P" (:eval (number-to-string (pdf-view-current-page)))
              ;; Avoid errors during redisplay.
              ":" (:eval (or (ignore-errors
                               (number-to-string (pdf-cache-number-of-pages)))
                             "???"))
	      "  PL" (:eval (pdf-view-current-pagelabel))))
```

If you're using doom-modeline you'd need to change `doom-modeline-format--pdf` instead. To make this permanent add the code to `pdf-view-mode-hook`.  You can also set a global `(setq my-pdf-offset 10)` and then add `(+ (pdf-view-current-page) my-pdf-offset)` instead.

**Comment by Twix53791** [comment]
Just to help the newbies like me to add this code to `pdf-view-mode-hook`, the code has to be put into a lambda function like that:

```lisp
(add-hook 'pdf-view-mode-hook
   (lambda ()
      (setq-local mode-line-position
            '(" P " (:eval (number-to-string (pdf-view-current-page)))
              ;; Avoid errors during redisplay.
              "/" (:eval (or (ignore-errors
                               (number-to-string (pdf-cache-number-of-pages)))
                             "???"))
            "  PL " (:eval (pdf-view-current-pagelabel)))
       )
   )
)
```

And if you want to keep the % scroll page indicator:

```lisp
(add-hook 'pdf-view-mode-hook
   (lambda ()
      (setq-local mode-line-position
            '((:eval (pdf-misc-size-indication))
              "  P " (:eval (number-to-string (pdf-view-current-page)))
              ;; Avoid errors during redisplay.
              "/" (:eval (or (ignore-errors
                               (number-to-string (pdf-cache-number-of-pages)))
                             "???"))
            "  PL " (:eval (pdf-view-current-pagelabel))))))
```
